packages/nextcloud/lib/src/helpers/common.dart

```diff
@@ -1,2 +1,43 @@
-/// The result of a version check.
-typedef VersionSupported<T> = ({bool isSupported, T minimumVersion});
+import 'package:meta/meta.dart';
+import 'package:version/version.dart';
+
+/// Holds the [versions], [minimumVersion] and [maximumVersion] of an app.
+@immutable
+class VersionCheck {
+  /// Creates a new [VersionCheck].
+  ///
+  /// If the [maximumVersion] is `null` the compatibility of the major of the [minimumVersion] is checked.
+  const VersionCheck({
+    required this.versions,
+    required this.minimumVersion,
+    required this.maximumVersion,
+  });
+
+  /// Current version of the app.
+  final Iterable<Version>? versions;
+
+  /// Minimum version of the app.
+  final Version minimumVersion;
+
+  /// Maximum version of the app.
+  final Version? maximumVersion;
```

Comment:
For server it would make sense to have a flag to set the maximum version to be exclusive. Then we can specify the max version as 28.0.0 for example instead of 27.99.99

Reply:
@Leptopoda what do you think?

Reply:
Github is garbage.
Because I replied from the review screen my comment is still "pending" :roll_eyes: 

---

Maybe we con do something with the minSupportedVersion.incrementMajor() and just check wether the version is smaller than the next major one.
Similar to what dynamite does.